src/register-org/containers/registered-address/registered-address.component.ts

```diff
@@ -22,14 +32,35 @@ export class RegisteredAddressComponent extends RegisterComponent implements OnI
 
   public ngOnInit(): void {
     super.ngOnInit();
+    this.setExistingFormGroup();
+  }
+
+  private setExistingFormGroup(): void {
+    if (this.registrationData.address && this.registrationData.address.addressLine1 && this.registrationData.address.addressLine1 !== '') {
+      if (this.registrationData.inInternationalMode) {
+        this.startedInternational = true;
+        this.isInternational = this.registrationData.address.country !== 'UK';
+        // in order to ensure user can go back even with persisted data
+        this.registrationData.inInternationalMode = false;
+        this.registerOrgService.persistRegistrationData(this.registrationData);
+      }
+      this.setFormGroup(this.registrationData.address);
+    }
   }
 
   public ngOnDestroy(): void {
     super.ngOnDestroy();
   }
 
   public onContinue(): void {
-    this.router.navigate([this.registerOrgService.REGISTER_ORG_NEW_ROUTE, 'document-exchange-reference']);
+    if (this.isFormValid()) {
+      this.registrationData.address = this.formGroup.get('address').value;
+      this.registrationData.inInternationalMode = this.startedInternational;
+      this.registerOrgService.persistRegistrationData(this.registrationData);
```

Comment:
@connorpgpmcelroy - persisting registration data will be automatically done by the register org service. You have to just call the `super.ngOnDestroy();` in your component's `ngOnDestroy` method.

Reply:
Ok, again will check and confirm that is the case and remove if so